src/fields/date/date.ts

```diff
@@ -50,6 +52,46 @@ export const date: IFieldGenerator<IDateSchema> = (id, { dateFormat = "uuuu-MM-d
 				.test("not-past", notPastRule?.["errorMessage"] || ERROR_MESSAGES.DATE.CANNOT_BE_PAST, (value) => {
 					if (!isValidDate(value, dateFormatter) || !notPastRule?.["notPast"]) return true;
 					return !LocalDate.parse(value, dateFormatter).isBefore(LocalDate.now());
+				})
+				.test("min-date", undefined, (value, context) => {
```

Comment:
the function should return true immediately if rule is not set to avoid unnecessary validation

actually looking back on these custom validations, `.test()` can also not be chained at all if the rule is not found

Reply:
ok have updated accordingly

yeah we need to optimise the way the custom validations are done, I think your approach is more optimised too, we shall revisit this when we migrate the libraries to the new repos